client/landing/stepper/declarative-flow/internals/components/step-route/test/index.tsx

```diff
@@ -0,0 +1,193 @@
+	getSignupCompleteStepNameAndClear,
+} from 'calypso/signup/storageUtils';
+import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
+import { renderWithProvider } from 'calypso/test-helpers/testing-library';
+import StepRoute from '../';
+import { NavigationControls } from '../../../types';
+
+jest.mock( 'calypso/signup/storageUtils' );
+jest.mock( 'calypso/state/current-user/selectors' );
+jest.mock( 'calypso/landing/stepper/declarative-flow/internals/analytics/record-step-start' );
+jest.mock( 'calypso/landing/stepper/hooks/use-intent' );
+jest.mock( 'calypso/landing/stepper/hooks/use-selected-design' );
+jest.mock( 'calypso/lib/analytics/page-view' );
+
+const RegularStep: StepperStep = {
+	slug: 'some-step-slug',
+	asyncComponent: jest.fn(), // For the purpose of this test, we don't need to render the using lazy strategy
+};
+
+const RequiresLoginStep: StepperStep = {
+	...RegularStep,
+	requiresLoggedInUser: true,
+};
+
+const FakeStep: FC< StepProps > = () => {
+	return <div>Step Content</div>;
+};
+
+const FakeFlow: Flow = {
+	name: 'some-flow',
+	useSteps: () => [ RegularStep ],
+	useStepNavigation: () => ( {
+		submit: () => {},
+	} ),
+
+	isSignupFlow: false,
+};
+
+const fakeRenderStep = ( step: StepperStep ) => {
+	const StepComponent = FakeStep;
+	const navigationControls = {} as NavigationControls;
+	const stepData = {} as any;
+
+	return (
+		<StepComponent
+			navigation={ navigationControls }
+			flow={ FakeFlow.name }
+			variantSlug={ FakeFlow.variantSlug }
+			stepName={ step.slug }
+			data={ stepData }
+		/>
+	);
+};
+
+interface RenderProps {
+	step: StepperStep;
+	renderStep?: ( step: StepperStep ) => JSX.Element | null;
+}
+
+const render = ( { step, renderStep = fakeRenderStep }: RenderProps ) => {
+	return renderWithProvider(
+		<MemoryRouter
+			basename="/setup"
+			initialEntries={ [ '/setup/some-flow/some-step-slug?param=example.com' ] }
+		>
+			<Suspense fallback={ null }>
+				<StepRoute
+					step={ step }
+					flow={ FakeFlow }
+					showWooLogo={ false }
+					renderStep={ renderStep }
+				/>
+			</Suspense>
+		</MemoryRouter>
+	);
+};
+
+describe( 'StepRoute', () => {
+	// we need to save the original object for later to not affect tests from other files
+	const originalLocation = window.location;
+
+	beforeAll( () => {
+		Object.defineProperty( window, 'location', {
+			value: { ...originalLocation, assign: jest.fn() },
+		} );
+		( useIntent as jest.Mock ).mockReturnValue( 'build' );
+		( useSelectedDesign as jest.Mock ).mockReturnValue( { design_type: 'premium' } );
+	} );
+
+	beforeEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	afterAll( () => {
+		Object.defineProperty( window, 'location', originalLocation );
+	} );
+
+	it( 'renders the step content', async () => {
+		const { getByText } = render( { step: RegularStep } );
+
+		expect( getByText( 'Step Content' ) ).toBeInTheDocument();
+	} );
+
+	it( 'doesnt the step content when render step returns null', async () => {
```

Comment:
Nit: we're missing a `render`.
```suggestion
	it( 'does not render the step content when renderStep() returns null', async () => {
```